### PR TITLE
[#94] ScheduleController 리팩터링

### PR DESCRIPTION
**Pull request by yanJuicy**
## 개요
- Schedule 리팩터링

### 1. 변경 전
Schedule Controller에서 쿼리 스트링(@RequestParam)의 종류에 따라서 메소드를 여러 개 만드려고 했음
### 변경 후
하나의 메소드 getSchedulesByRequestParm에서 3가지 종류의 쿼리 스트링을 다 받고 종류에 따라 서비스 메서드 호출
### 2. 변경 전
쿼리 스트링 조건에 따라서 리턴해주는 레코드 객체가 달랐음
- theaterId 요청 -> ResponseFindMovieListAndDateList 반환
- theaterId, date 요청 -> ResponseFindMovieList 반환
- theaterId, movieId, date 요청 -> ResponseTimeList 반환
### 변경 후
ResponseFindSchedule 객체만 리턴해주기로 변경
- 위 객체 안에 movieList, theaterList, dateList, timeList 가 들어 있음
- 컨트롤러에서 쿼리 스트링 조건에 따라서 메소드를 분리할 수 없어서 위와 같이 변경을 결정함
### 3. 변경 전
theaterId로 요청해서 movieList, dateList를 반환 받는 API 문서의 변경
[이전 문서](https://github.com/prgrms-be-devcourse/BE-02-MarBox/pull/86)
### 변경 후
한 영화관의 영화, 날짜 정보 가져오기
<h3 id="_한_영화관의_영화_날짜_정보_가져오기" style="box-sizing: border-box; margin: 1em 0px 0.5em; padding: 0px; font-family: &quot;Open Sans&quot;, &quot;DejaVu Sans&quot;, sans-serif; font-weight: 300; font-style: normal; color: rgb(204, 204, 204); text-rendering: optimizelegibility; line-height: 1.2; font-size: 1.6875em; word-spacing: -0.05em; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; -webkit-text-stroke-width: 0px; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">한 영화관의 영화, 날짜 정보 가져오기</h3><div class="listingblock has-source-line data-line-/Users/hanju/IdeaProjects/BE-02-MarBox/build/generated-snippets/schedule-get-movie-and-date-in-theater/http-request.adoc-2" style="box-sizing: border-box; margin: 0px 0px 1.25em; padding: 0px; color: rgb(187, 187, 187); font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="title" style="box-sizing: border-box; margin: 0px 0px 0.25em; padding: 0px; line-height: 1.45; color: rgb(122, 37, 24); font-weight: 400; text-rendering: optimizelegibility; text-align: left; font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 1rem; font-style: italic; filter: invert(90%) hue-rotate(180deg);">Request</div><div class="content" style="box-sizing: border-box; margin: 0px; padding: 0px; position: relative;"><pre class="CodeRay highlight nowrap" style="box-sizing: border-box; font-family: &quot;DejaVu Sans Mono&quot;, monospace; font-size: 0.90625em; white-space: pre; margin: 0px; padding: 1em; color: rgba(0, 0, 0, 0.9); line-height: 1.45; text-rendering: optimizespeed; overflow-wrap: normal; background: rgb(247, 247, 248); filter: invert(90%) hue-rotate(180deg); border-radius: 4px; overflow-x: auto;"><code data-lang="http" style="box-sizing: border-box; font-family: &quot;DejaVu Sans Mono&quot;, monospace; font-size: inherit; font-weight: 400; color: inherit; line-height: inherit; display: block;">GET /schedules?theaterId=1 HTTP/1.1
Host: localhost:8080</code></pre></div></div><div class="listingblock has-source-line data-line-/Users/hanju/IdeaProjects/BE-02-MarBox/build/generated-snippets/schedule-get-movie-and-date-in-theater/http-response.adoc-2" style="box-sizing: border-box; margin: 0px 0px 1.25em; padding: 0px; color: rgb(187, 187, 187); font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="title" style="box-sizing: border-box; margin: 0px 0px 0.25em; padding: 0px; line-height: 1.45; color: rgb(122, 37, 24); font-weight: 400; text-rendering: optimizelegibility; text-align: left; font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 1rem; font-style: italic; filter: invert(90%) hue-rotate(180deg);">Response</div><div class="content" style="box-sizing: border-box; margin: 0px; padding: 0px; position: relative;"><pre class="CodeRay highlight nowrap" style="box-sizing: border-box; font-family: &quot;DejaVu Sans Mono&quot;, monospace; font-size: 0.90625em; white-space: pre; margin: 0px; padding: 1em; color: rgba(0, 0, 0, 0.9); line-height: 1.45; text-rendering: optimizespeed; overflow-wrap: normal; background: rgb(247, 247, 248); filter: invert(90%) hue-rotate(180deg); border-radius: 4px; overflow-x: auto;"><code data-lang="http" style="box-sizing: border-box; font-family: &quot;DejaVu Sans Mono&quot;, monospace; font-size: inherit; font-weight: 400; color: inherit; line-height: inherit; display: block;">HTTP/1.1 200 OK
Content-Type: application/json
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
X-Frame-Options: DENY
Content-Length: 420

{"movieList":[{"name":"영화1","limitAge":"CHILD","genre":"ACTION","runningTime":100,"posterImgLocation":"test/location"},{"name":"영화2","limitAge":"ADULT","genre":"ROMANCE","runningTime":120,"posterImgLocation":"test/location"},{"name":"영화3","limitAge":"CHILD","genre":"ANIMATION","runningTime":150,"posterImgLocation":"test/location"}],"theaterList":null,"dateList":["2022-06-30","2022-07-01"],"timeList":null}</code></pre></div></div>

Path | Type | Description
-- | -- | --
movieList[] | Array | 영화 리스트
movieList[].name | String | 영화 이름
movieList[].limitAge | String | 영화 관람 등급
movieList[].genre | String | 장르
movieList[].runningTime | Number | 상영시간
movieList[].posterImgLocation | String | 포스터 이미지 경로
dateList[] | Array | 상영 날짜 리스트
theaterList | Null | 영화관 리스트
timeList | Null | 상영 시간 리스트

## 기타
API 문서에 null 값을 빈 배열 [] 로 할지 null 로 할지 고민이 됩니다

### CODE REVIEW

File: src/test/java/prgrms/marco/be02marbox/domain/theater/service/ScheduleServiceTest.java [hunk 1]
```diff
@@ -121,7 +123,7 @@ void testGetCurrentMovieList_No_Duplicate() {
 		createAndSaveSchedule(theaterRoom2, movie, LocalDateTime.now().plusDays(2), LocalDateTime.now());
 		createAndSaveSchedule(theaterRoom, movie, LocalDateTime.now().plusDays(4), LocalDateTime.now());
 
-		assertThat(scheduleService.findCurrentMovieList().size()).isEqualTo(1);
+		assertThat(scheduleService.findShowingMovieList().size()).isEqualTo(1);
 	}
```
Comment (wisehero) [inline src/test/java/prgrms/marco/be02marbox/domain/theater/service/ScheduleServiceTest.java:127]: 리스트의 크기의 경우 hasSize()가 조금 더 직관적인 것 같습니다!

**Review by wisehero — APPROVED** [review]
LGTM

File: src/test/java/prgrms/marco/be02marbox/domain/theater/controller/ScheduleControllerTest.java [hunk 2]
```diff
@@ -125,22 +128,45 @@ void testGetCurrentMovieList() throws Exception {
...
+	@Test
+	@DisplayName("존재하지 않는 영화관 ID로 요청하면 404 Not Found")
+	@WithMockUser(roles = {"ADMIN", "USER"})
+	void testGetMovieListAndDateListInOneTheater_Fail_Invalid_Theater() throws Exception {
+		given(scheduleService.findMovieListAndDateListByTheaterId(200L))
+			.willThrow(new EntityNotFoundException(Message.INVALID_THEATER_EXP_MSG.getMessage()));
+
+		mockMvc.perform(get("/schedules")
+				.param("theaterId", "200"))
+			.andExpect(status().isNotFound());
```
Comment (hyuk0309) [inline src/test/java/prgrms/marco/be02marbox/domain/theater/controller/ScheduleControllerTest.java:169]: jsonPath를 이용하셔서 응답 바디 값도 같이 검증하는 테스트 작성하시면 가독성이 좋아질 것 같습니다! 🙋‍♂️

**Review by hyuk0309 — COMMENTED** [review]
고생하셨습니다! 👏

응답 Dto를 하나로 통일하셨는데, 상속 구조를 이용하면 컨트롤러 하나로 여러개의 Dto를 반환할 수 있지 않을까요? 🤔 (뇌피셜입니당..ㅎㅎ)

File: src/test/java/prgrms/marco/be02marbox/domain/theater/controller/ScheduleControllerTest.java (hunk 2 above)
Comment (yanJuicy) [reply src/test/java/prgrms/marco/be02marbox/domain/theater/controller/ScheduleControllerTest.java:169]: 상속 한 번 고민해봐야겠네요!

**Review by hyuk0309 — APPROVED** [review]
고생하셨슴돠! 👏

**Pull request merged**